hack/update-openapi-spec.sh

```diff
@@ -96,10 +96,21 @@ if ! kube::util::wait_for_url "https://${API_HOST}:${API_PORT}/healthz" "apiserv
   exit 1
 fi
 
-kube::log::status "Updating " "${OPENAPI_ROOT_DIR}"
+kube::log::status "Updating " "${OPENAPI_ROOT_DIR} for OpenAPI v2"
 
 curl -w "\n" -kfsS -H 'Authorization: Bearer dummy_token' "https://${API_HOST}:${API_PORT}/openapi/v2" | jq -S '.info.version="unversioned"' > "${OPENAPI_ROOT_DIR}/swagger.json"
 
+kube::log::status "Updating " "${OPENAPI_ROOT_DIR}/v3 for OpenAPI v3"
```

Comment:
I've often wondered why we produce these.  Do we really want to generate these?  I'd be happy to see them removed and/or generated from the source instead of the server.

Reply:
I think it'd be great if we could generate from the source rather than the server, on the other hand, it's super useful to be able to see what changed or that nothing changed.

Reply:
I don't have a strong opinion about this and am happy to remove if needed. (It'll reduce the likelihood of merge conflicts as well). 

OTOH the `swagger.json` generated for v2 is used in some tests, but v3 has no such dependency.

Reply:
+1 for direction

Reply:
v2 spec is used for [generating clients](https://github.com/kubernetes-client/gen/blob/291f9a6ca5004cc91cfc4a744abf129c639d5682/openapi/preprocess_spec.py#L517-L521). Potentially we want to use a v3 equivalent in the future